diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5f644db..d3238d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,9 +132,15 @@ if(CAMERAADAPTOR_ENABLED AND RECORDER_SERVICE_ENABLED AND MEMORY_ENABLED)
   else()
     # Mainline build
     # Camx include paths are defined in common config
-    build_target("_kodiak" "qcm6490" "${CAMX_KODIAK_INCLUDE_DIRS}")
-    # Using lemans config as common target
-    build_target("" "qcs9100" "${CAMX_TARGET_INCLUDE_DIRS}")
+    if(DEFINED CAMX_KODIAK_INCLUDE_DIRS)
+      build_target("_kodiak" "qcm6490" "${CAMX_KODIAK_INCLUDE_DIRS}")
+    endif()
+
+    # Using qcs-common config
+    if(DEFINED CAMX_TARGET_INCLUDE_DIRS)
+      build_target("" "qcs-common" "${CAMX_TARGET_INCLUDE_DIRS}")
+    endif()
+
   endif()
 
 endif()
